### DISCUSSION TITLE
New MeshLab 2021.10 version!

### DISCUSSION THREAD

**Discussion by alemuntoni** [cnr-isti-vclab/meshlab#1135] [General]
This version presents redesigned filter dialogs (with scrollbars when necessary), some reorganization of filters (new, removed and renamed filters), updated libraries and lot of bugfixes. 

New filter plugins are:
- Nxs build
- Nxs compress
- Overlapping meshes
- ICP between meshes
- Global align meshes
- Define new per vertex custom point attribute
- Define new per face custom point attribute

Removed old filters:
- change the current layer
- change visibility of layers
- compact faces
- compact vertices
- cross field creation
- select folded faces
- vertex color noise

Unified filters:
- repair non manifold edges (will allow to select which method use)
- remove t-vertices (will allow to select which method use)

Along with MeshLab 2021.10, we also released [PyMeshLab 2021.10](https://pypi.org/project/pymeshlab/) ([here](https://pymeshlab.readthedocs.io/en/latest/changelog.html)  the changelog). Starting from the next pymeshlab version, almost all the pymeshlab filters will be renamed according to what is the main goal of each filter, making more clear what a filter does.
We will provide scripts that will replace old names with new names automatically, in order to make this change less painful as possible.

Enjoy this new version! :)

**Comment by eithor** [comment]
Is it a bug or by design that the identifier has changed from com.vcg.meshlab to com.vcg.nxsview ?

**Reply by alemuntoni** [reply]
It is definitely a bug. Thanks for reporting it!!

**Comment by Mikhail-Kozlov-MRI** [comment]
Would you, please, explain difference between
MeshLab2021.10-windows.zip
and
MeshLab2021.10d-windows.zip

Another words, what additionally included in "d" version?

**Reply by alemuntoni** [reply]
It is an additional version that we release since 2020.12 (#843). It stores floating point values with double precision, and it is the version that is used by pymeshlab.

**Comment by Luvideria** [comment]
This version of meshlab seem to produce unintended results with certain meshes. http://casual-effects.com/data/ breakfast room scene.
unintended behavior: the texture from the painting becomes the texture for the table

Maybe it's just me ? I downloaded a fresh copy of meshlab, and a fresh scene, it seems reproducible from my end.
Blender and the microsoft 3D Viewer both work as intended (the painting at the correct place)
Note that this file also crashed my previous version fo meshlab, but I do not remember which one.

**Reply by rpavlik** [reply]
I can confirm this, the textures are all messed up in 2022.02 as well:
![image](https://user-images.githubusercontent.com/61129/191315951-ab32f3f0-baba-415a-be0a-d114da15ab2c.png)
